spec/lib/jets/dotenv_spec.rb

```diff
@@ -0,0 +1,63 @@
+describe Jets::Dotenv do
+  describe "#load!" do
+    it "replaces ${ssm:<relative-path>} with SSM parameters prefixed with /<app-name>/<jets-env>/" do
+      expect(::Dotenv).to receive(:load).and_return(
+        "AUTENTICATED_URL" => "https://${ssm:username}:${ssm:password}@host.com",
+      )
+
+      username_response_double = double(
+        Aws::SSM::Types::GetParameterResult,
+        parameter: Aws::SSM::Types::Parameter.new(value: "my-user"),
+      )
+      expect_any_instance_of(Aws::SSM::Client).to receive(:get_parameter)
+        .with(name: "/demo/test/username", with_decryption: true)
+        .and_return(username_response_double)
```

Comment:
I'm not a big fan of all the detailed mocking required in the specs.

I was originally surprised by these calls already being stubbed via `Aws.config.update(stub_responses: true)`. Maybe there are more conventional ways of mocking the `Aws` libraries`?

Reply:
Yeah, mocking is of work. Think this is fine, it works 👍 Note, for specs, usually I ask myself, is it saving me overall development time.